scapy/automaton.py

```diff
@@ -34,199 +35,105 @@
-    remain: timeout. If 0, return [].
-    customTypes: types of the objects that have the check_recv function.
-    """
+        def close(self):
+            pass
 
-    def _release_all(self):
-        """Releases all locks to kill all threads"""
-        for i in self.inputs:
-            i.call_release(True)
-        self.available_lock.release()
-
-    def _timeout_thread(self, remain):
-        """Timeout before releasing every thing, if nothing was returned"""
-        time.sleep(remain)
-        if not self._ended:
-            self._ended = True
-            self._release_all()
-
-    def _exit_door(self, _input):
-        """This function is passed to each SelectableObject as a callback
-        The SelectableObjects have to call it once there are ready"""
-        self.results.append(_input)
-        if self._ended:
-            return
-        self._ended = True
-        self._release_all()
-
-    def __init__(self, inputs, remain):
-        self.results = []
-        self.inputs = list(inputs)
-        self.remain = remain
-        self.available_lock = threading.Lock()
-        self.available_lock.acquire()
-        self._ended = False
-
-    def process(self):
-        """Entry point of SelectableSelector"""
-        if WINDOWS:
-            select_inputs = []
-            for i in self.inputs:
-                if not isinstance(i, SelectableObject):
-                    warning("Unknown ignored object type: %s", type(i))
-                elif i.__selectable_force_select__:
-                    # Then use select.select
-                    select_inputs.append(i)
-                elif not self.remain and i.check_recv():
-                    self.results.append(i)
-                elif self.remain:
-                    i.wait_return(self._exit_door)
-            if select_inputs:
-                # Use default select function
-                self.results.extend(select(select_inputs, [], [], self.remain)[0])  # noqa: E501
-            if not self.remain:
-                return self.results
-
-            threading.Thread(
-                target=self._timeout_thread,
-                args=(self.remain,),
-                name="scapy.automaton process"
-            ).start()
-            if not self._ended:
-                self.available_lock.acquire()
-            return self.results
-        else:
-            r, _, _ = select(self.inputs, [], [], self.remain)
-            return r
+    def check_recv(self):
+        return False
 
 
 def select_objects(inputs, remain):
     """
     Select SelectableObject objects. Same than:
-    ``select.select([inputs], [], [], remain)``
+    ``select.select(inputs, [], [], remain)``
     But also works on Windows, only on SelectableObject.
 
     :param inputs: objects to process
     :param remain: timeout. If 0, return [].
     """
-    handler = SelectableSelector(inputs, remain)
-    return handler.process()
-
-
-class ObjectPipe(SelectableObject, io.BufferedIOBase):
-
+    if WINDOWS:
```

Comment:
I'd rather handle the simple case first, then the Windows case:
Rather than:
```Python
    if WINDOWS:
        natives = []
        [... complex code ...]
    else:
        return select.select(inputs, [], [], remain)[0]
```
What do you think of:
```Python
    if not WINDOWS:
        return select.select(inputs, [], [], remain)[0]
    natives = []
    [... complex code ...]
```